From 6c947c5b50eb9a8427de82b7499594d180fd945b Mon Sep 17 00:00:00 2001
From: Mathieu Bridon <bochecha@daitauha.fr>
Date: Sun, 18 Feb 2018 14:45:31 +0100
Subject: [PATCH] Find libcuda.so.1

This is a backport of an upstream patch, to get CUDA to work with our
Flatpak build.

    From d3829411b6e006cfe1b031c6a09ae63cf7878522 Mon Sep 17 00:00:00 2001
    From: Brecht Van Lommel <brechtvanlommel@gmail.com>
    Date: Sun, 18 Feb 2018 12:57:42 +0100
    Subject: [PATCH] Search for libcuda.so.1 as well, not all Linux distributions have libcuda.so.

    https://github.com/CudaWrangler/cuew/pull/6

It can be removed once the upstream patch made it into a Blender
release.
---
 extern/cuew/src/cuew.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extern/cuew/src/cuew.c b/extern/cuew/src/cuew.c
index c96ea2c..e2cc4b9 100644
--- a/extern/cuew/src/cuew.c
+++ b/extern/cuew/src/cuew.c
@@ -323,7 +323,7 @@ int cuewInit(void) {
   const char *cuda_paths[] = {"/usr/local/cuda/lib/libcuda.dylib", NULL};
   const char *nvrtc_paths[] = {"/usr/local/cuda/lib/libnvrtc.dylib", NULL};
 #else
-  const char *cuda_paths[] = {"libcuda.so", NULL};
+  const char *cuda_paths[] = {"libcuda.so", "libcuda.so.1", NULL};
   const char *nvrtc_paths[] = {"libnvrtc.so",
 #  if defined(__x86_64__) || defined(_M_X64)
                                "/usr/local/cuda/lib64/libnvrtc.so",
-- 
2.14.3

